A100-client-side-weighted-round-robin-slow-start.md

```diff
@@ -0,0 +1,254 @@
+  // The value of aggression parameter should be greater than 0.0.
+  // By tuning the parameter, it is possible to achieve polynomial or exponential shape of ramp-up curve.
+  //
+  // During slow start window, effective weight of an endpoint would be scaled with time factor and aggression:
+  // ``new_weight = weight * max(min_weight_percent / 100, time_factor ^ (1 / aggression))``,
+  // where ``time_factor=(time_since_start_seconds / slow_start_time_seconds)``.
+  //
+  // As time progresses, more and more traffic would be sent to endpoint, which is in slow start window.
+  // Once host exits slow start, time_factor and aggression no longer affect its weight.
+  google.protobuf.FloatValue aggression = 2;
+
+  // Configures the minimum percentage of the original weight that will be used for an endpoint
+  // in slow start. This helps to avoid a scenario in which endpoints receive no traffic during the
+  // slow start window. Must be between 0 and 100. If not specified, the default is 10%.
+  google.protobuf.UInt32Value min_weight_percent = 3;
+}
+```
+
+### Weight Scaling During Warmup
+
+When an endpoint is ready after being in a non-ready state, it enters the warmup period. During this period, its weight will be scaled by a factor that increases non-linearly from `min_weight_percent` to 100% over the duration of `slow_start_window`.
+
+The scale factor is calculated as follows:
+
+```
+time_factor = max(time_since_ready_in_seconds, 1) / slow_start_window_seconds
+scale = max(min_weight_percent/100, time_factor ^ (1/aggression))
+```
+
+The following image shows how different aggression values affect the scaling factor over time during the slow start window (in milliseconds):
+
+![Effect of aggression parameter on slow start scaling](A123_graphics/aggression_scaling.png)
+
+The graph illustrates how the scaling factor (effective weight) ramps up from zero to the backend weight for various aggression values, with time shown in milliseconds.
+
+The final weight used for load balancing will be:
+```
+final_weight = computed_weight * scale
+```
+
+### Subchannel Weights
+
+To maintain independence between the blackout period and slow start period, a new timestamp `ready_since` will be introduced to track when an endpoint transitioned to ready state. This timestamp is separate from the existing `non_empty_since` timestamp used for the blackout period.
+
+The `ready_since` timestamp is set when an endpoint transitions from a non-ready state (e.g., CONNECTING, TRANSIENT_FAILURE) to a ready state (READY). This marks the beginning of the slow start period for that endpoint. The existing `non_empty_since` timestamp continues to be used exclusively for tracking the blackout period, which begins when the first non-zero load report is received.
+
+Weight calculation in the WRR policy follows a two-step process. First, the base weight for each endpoint is computed using the formula from [gRFC A58][A58]:
```

Comment:
This part seems to basically repeat what is already covered in the previous section.  I don't think we need to spell out the formula twice.  Maybe just remove the previous section?